torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3578,7 +3581,7 @@ def _cast_grad_to_param_dtype(
             grad.data = grad.data.to(dtype=param.dtype)
             # Do not let the low precision gradient memory get reused until
             # the cast to full parameter precision completes
-            low_prec_grad_data.record_stream(torch.cuda.current_stream())
+            low_prec_grad_data.record_stream(self._streams["post_backward"])
```

Comment:
Curious if we can clearly show a proof that this is correct in all cases (or if there's another way to validate correctness). Changing stream semantics and testing correctness is a bit tricky because issues / race conditions may only rarely reproduce.